### ISSUE TITLE
How to add the server url in the image src?

### ISSUE THREAD

**Issue by Lurtz59** [Roslovets-Inc/strapi-plugin-ckeditor5#12]
Hello,

Thanks for this wonderful plugin :)

I want to display the uploaded images on my site but I cannot. Because the URL path is wrong. I get something like "/uploads/test.png".

How can I change the path to put my server path before "/ uploads /" please?

**Comment by Mites88** [comment]
+1 I'm on the same boat.

**Comment by Mites88** [comment]
After inspecting the source code for a while, I believe [this line](https://github.com/Roslovets-Inc/strapi-plugin-ckeditor5/blob/9e87a03aa9716ceba7232cf5ce890fd36654885b/admin/src/components/Wysiwyg/index.js#L47) is our culprit.

``prefixFileUrlWithBackendUrl`` isn't working on my build (strapi version 3.5.3)

If I replace   ``const url = prefixFileUrlWithBackendUrl(data.url);`` with ``const url = `http://localhost:1337${data.url}`;``, it will work.

``${strapi.backendUrl}`` seems to not be working on my system too, dev and prod builds.

**Comment by Lurtz59** [comment]
Hello,

Thank you for your reply!
Unfortunately, it doesn't work :(

When I get a strapi image I always have "/uploads/..." instead of having "http://localhost:20001/uploads/...".
I don't know if it's possible to make this work.

**Comment by roslovets** [comment]
Hi! The plugin uses [ckeditor5-strapi-upload-plugin](https://github.com/gtomato/ckeditor5-strapi-upload-plugin) to work with images. Can the problem be on its side?

**Comment by Lurtz59** [comment]
Hi !

I don't know. It's possible. I want to display the images of ckeditor on my site.
![image](https://user-images.githubusercontent.com/58726714/111869250-b9bd2f00-897e-11eb-8bfd-32023c055af1.png)

Like that : 
![image](https://user-images.githubusercontent.com/58726714/111869280-e3765600-897e-11eb-91e4-db98e0c0909b.png)

**Comment by roslovets** [comment]
Seems like it's default Strapi behaviour. Does default WISYWIG editor provide full image URL?

**Comment by roslovets** [comment]
BTW don't you forget to rebuild Strapi after [this trick](https://github.com/Roslovets-Inc/strapi-plugin-ckeditor5/issues/12#issuecomment-803243998)?

**Comment by Mites88** [comment]
> Seems like it's default Strapi behaviour. Does default WISYWIG editor provide full image URL?

``const url = prefixFileUrlWithBackendUrl(data.url);``

This function seems to return nothing if you don't have an ``url`` set on your server object.

**Comment by Mites88** [comment]
> Hello,
> 
> Thank you for your reply!
> Unfortunately, it doesn't work :(
> 
> When I get a strapi image I always have "/uploads/..." instead of having "http://localhost:20001/uploads/...".
> I don't know if it's possible to make this work.

Please try rebuilding your admin and also forcing a refresh on your CKEditor page(``CTRL + R`` on Chrome), I had some caching issues.

**Comment by Lurtz59** [comment]
I just rebuilt and clear the cache it doesn't work. 
When I use ckeditor to host an image in a rich text field and  i interpret that rich field on my forehead. I get "/uploads/" all the time.

My server config : 
![image](https://user-images.githubusercontent.com/58726714/111869815-61d3f780-8981-11eb-9b19-5004c1b68c54.png)

My rich text : 
![image](https://user-images.githubusercontent.com/58726714/111869839-7ca66c00-8981-11eb-8234-aa1dc1b1fe52.png)

**Comment by Mites88** [comment]
> I just rebuilt and clear the cache it doesn't work.
> When I use ckeditor to host an image in a rich text field and i interpret that rich field on my forehead. I get "/uploads/" all the time.
> 
> My server config :
> ![image](https://user-images.githubusercontent.com/58726714/111869815-61d3f780-8981-11eb-9b19-5004c1b68c54.png)
> 
> My rich text :
> ![image](https://user-images.githubusercontent.com/58726714/111869839-7ca66c00-8981-11eb-8234-aa1dc1b1fe52.png)
> 
> I am using 'v-html' to interpret the html text I get from strapi.
> 
> **EDIT: I think I will use a 'computed' to fix this problem. Anyway, thanks for your help!**

Yup, your config seems correct.

As a workaround, you can hardcode [this line](https://github.com/Roslovets-Inc/strapi-plugin-ckeditor5/blob/9e87a03aa9716ceba7232cf5ce890fd36654885b/admin/src/components/Wysiwyg/index.js#L47) until we figure out a better fix.

**Comment by roslovets** [comment]
This issue can be related
https://github.com/strapi/strapi/issues/9314

**Comment by roslovets** [comment]
Seems like answer
https://github.com/strapi/strapi/issues/9176#issuecomment-765695884

**Comment by Mites88** [comment]
I appreciate your help investigating the issue, @roslovets.
As far as I understand, there's very little documentation to the strapi core, which makes word of mouth much more powerful.

**Comment by harish2704** [comment]
IMHO, this is an bug caused by one of the[ indirect upstream dependency of this project ](https://github.com/gtomato/ckeditor5-strapi-upload-plugin). 

So, to resolve this issue, 

1. First , the package https://github.com/gtomato/ckeditor5-strapi-upload-plugin should to add an option to specify  `uploaded_media_prefix` ( for eg: `media_prefix` in addition to `uploadUrl` option )
2. Then,  the package https://github.com/Roslovets-Inc/ckeditor5-build-strapi-wysiwyg ( which is a direct dependency of this project ) need to specify the above mentioned option while initializing Ckeditor widget .  Since it is a pre-compiled dependecy,  After the update, it should be re-build and release the package .
3. After that, This package need to update the depenedency version and use the bug fixed version of the ckeditor build

I think, it will take time for all these things to happen, so I did a quick fix, 

Ie, derive the `media_prefix` from `uploadUrl` and use it in the https://github.com/gtomato/ckeditor5-strapi-upload-plugin .

I did that, rebuiled the ck editor and [published the release](https://github.com/harish2704/ckeditor5-build-strapi-wysiwyg/releases/tag/v1.13.0-fixurlprefix). 

For those who got stuck at this issue, a quick but dirty solution will be to add the following to package.json
```json
  "resolutions": {
    "ckeditor5-build-strapi-wysiwyg": "https://github.com/harish2704/ckeditor5-build-strapi-wysiwyg/releases/download/v1.13.0-fixurlprefix/ckeditor5-build-strapi-wysiwyg-v1.13.0.tgz"
  },
```

**Comment by gabriel-henriq** [comment]
Any updates on this issue?

**Comment by anjum121** [comment]
It's working with follow steps.
-  Add below code into ```/config/server.js```
````javascript
module.exports = ({ env }) => ({
  host: env("HOST", "0.0.0.0"),
  port: env.int("PORT", 1337),
  url: env("PUBLIC_URL", "http://localhost:1337"),
});
````
-  Rebuild strapi 
```` 
npm run build 
````
-  Refresh browser with cache clearing  
-  When you add new image try using body button.
<img width="926" alt="Screen Shot 2022-02-06 at 8 15 38 PM" src="https://user-images.githubusercontent.com/1026657/152690341-01fc9d21-8860-4b9c-9192-6a50dfadf860.png">

**Comment by Blazing-Mike** [comment]
@anjum121 what should the PUBLIC_URL be in this case?

**Comment by Blazing-Mike** [comment]
> IMHO, this is an bug caused by one of the[ indirect upstream dependency of this project ](https://github.com/gtomato/ckeditor5-strapi-upload-plugin).
> 
> So, to resolve this issue,
> 
> 1. First , the package https://github.com/gtomato/ckeditor5-strapi-upload-plugin should add an option to specify  `uploaded_media_prefix` ( for eg: `media_prefix` in addition to `uploadUrl` option )
> 2. Then,  the package https://github.com/Roslovets-Inc/ckeditor5-build-strapi-wysiwyg ( which is a direct dependency of this project ) need to specify the above mentioned option while initializing Ckeditor widget .  Since it is a pre-compiled dependecy,  After the update, it should be re-build and release the package .
> 3. After that, This package need to update the depenedency version and use the bug fixed version of the ckeditor build
> 
> I think, it will take time for all these things to happen, so I did a quick fix,
> 
> Ie, I derived the `media_prefix` from `uploadUrl` and use it in the https://github.com/gtomato/ckeditor5-strapi-upload-plugin .
> 
> I did that, rebuiled the ckeditor and [published the release](https://github.com/harish2704/ckeditor5-build-strapi-wysiwyg/releases/tag/v1.13.0-fixurlprefix).
> 
> For those who got stuck at this issue, a quick but dirty solution will be to add the following to your `package.json`.
> 
> ```json
>   "resolutions": {
>     "ckeditor5-build-strapi-wysiwyg": "https://github.com/harish2704/ckeditor5-build-strapi-wysiwyg/releases/download/v1.13.0-fixurlprefix/ckeditor5-build-strapi-wysiwyg-v1.13.0.tgz"
>   },
> ```

@harish2704 does adding this resolution to package.json fix the img src issue

**Comment by harish2704** [comment]
> @harish2704 does adding this resolution to package.json fix the img src issue

@Blazing-Mike : Yes. You can verify this by adding my custom buid as a dependency as I mentioned in the end of my first message

**Comment by Blazing-Mike** [comment]
@harish2704 i am quite a noob in strapi... What i need clarity on is does add this code block to my package.json solve the issue?

```
  "resolutions": {
    "ckeditor5-build-strapi-wysiwyg": "https://github.com/harish2704/ckeditor5-build-strapi-wysiwyg/releases/download/v1.13.0-fixurlprefix/ckeditor5-build-strapi-wysiwyg-v1.13.0.tgz"
  },
```

**Comment by harish2704** [comment]
> @harish2704 i am quite a noob in strapi... What i need clarity on is does add this code block to my package.json solve the issue?
> 
> ```
>   "resolutions": {
>     "ckeditor5-build-strapi-wysiwyg": "https://github.com/harish2704/ckeditor5-build-strapi-wysiwyg/releases/download/v1.13.0-fixurlprefix/ckeditor5-build-strapi-wysiwyg-v1.13.0.tgz"
>   },
> ```

@Blazing-Mike : Yes bro. It resolved the issue at the time of my first comment. I am not sure about the current situation . 
But you can verify this by simply trying out

**Comment by Makskokhtenko** [comment]
I found the answer on the CKEditor Plugin page, so you just need to do this:

If you use default upload provider and you want prefix img url with api path you need to add baseURL in api.js file (config/api.js)

`module.exports = {
  rest: {
    defaultLimit: 25,
    maxLimit: 100,
    withCount: true,
  },
  baseURL: 'http://localhost:1337'
};`

Then rebuild and restart

**Comment by pmeet862** [comment]
you have to re-upload the image after doing this.

> I found the answer on the CKEditor Plugin page, so you just need to do this:
> 
> If you use default upload provider and you want prefix img url with api path you need to add baseURL in api.js file (config/api.js)
> 
> `module.exports = { rest: { defaultLimit: 25, maxLimit: 100, withCount: true, }, baseURL: 'http://localhost:1337' };`
> 
> Then rebuild and restart

you have to re-upload the image after doing this.

**Comment by pmeet862** [comment]
> I found the answer on the CKEditor Plugin page, so you just need to do this:
> 
> If you use default upload provider and you want prefix img url with api path you need to add baseURL in api.js file (config/api.js)
> 
> `module.exports = { rest: { defaultLimit: 25, maxLimit: 100, withCount: true, }, baseURL: 'http://localhost:1337' };`
> 
> Then rebuild and restart

you have to re-upload the image after doing this.

**Comment by gilzonme** [comment]
Adding the `url` parameter with public url in `config/server.js` solves the issue

```
module.exports = ({ env }) => ({
  url:"https://cms.example.com",
  host: env('HOST', '0.0.0.0'),
  port: env.int('PORT', 1337),
  app: {
    keys: env.array('APP_KEYS'),
  },
});
```

**Comment by kirwako** [comment]
you can use this strapi plugin
that prepend public url/media prefix to your media URL in api's URL
it works well and it will your problem
it converts the URLs from `/uploads/*` to `https://your-media-link.com/uploads/*`
https://www.npmjs.com/package/strapi-plugin-media-prefix
